float32 leftEnc
float32 rightEnc